assets/blocks/course-categories-block/course-categories-edit.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import { unescape } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useBlockProps } from '@wordpress/block-editor';
+import { Spinner } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { useMemo } from 'react';
+import useCourseCategories from './hooks/use-course-categories';
+import {
+	withColorSettings,
+	withDefaultColor,
+} from '../../shared/blocks/settings';
+import { compose } from '@wordpress/compose';
+
+export function CourseCategoryEdit( props ) {
+	const { context, attributes, textColor, backgroundColor } = props;
+	const { textAlign } = attributes;
+	const { postId } = context;
+	const term = 'course-category';
+
+	const {
+		postTerms: categories,
+		hasPostTerms: hasCategories,
+		isLoading,
+	} = useCourseCategories( postId );
+
+	const blockProps = useBlockProps( {
+		className: classnames( {
+			[ `has-text-align-${ textAlign }` ]: textAlign,
+			[ `taxonomy-${ term }` ]: term,
+		} ),
+	} );
+
+	const inlineStyle = useMemo(
+		() => ( {
+			color: textColor?.color,
+			backgroundColor: backgroundColor?.color,
+		} ),
+		[ textColor, backgroundColor ]
+	);
+
+	return (
+		<>
+			<div { ...blockProps }>
+				{ isLoading && <Spinner /> }
+				{ ! isLoading &&
+					hasCategories &&
+					categories.map( ( category ) => (
+						<a
+							key={ category.id }
+							href={ category.link }
+							onClick={ ( event ) => event.preventDefault() }
+							style={ inlineStyle }
+						>
+							{ unescape( category.name ) }
+						</a>
+					) ) }
+			</div>
+		</>
+	);
+}
+
+export default compose(
+	withColorSettings( {
+		textColor: {
+			style: 'color',
+			label: __( 'Text color', 'sensei-lms' ),
+		},
+		backgroundColor: {
+			style: 'background-color',
+			label: __( 'Category background color', 'sensei-lms' ),
+		},
+	} ),
+	withDefaultColor( {
```

Comment:
Hey @gabrielcaires! I'm really sorry for the delay to check this!

I focused my eyes on this part of the PR, related to the colors, and in the backend rendering of the color. It seems good to me.

But another approach I could suggest here since we're already using an `<a>` is, instead of the probe approach, to use something like: `.my-link { background-color: currentColor; }`. The con of that is that we wouldn't have the exact `color` property from the theme. But if it's okay for this project, we could set white or black, depending on what could be more readable with the `background-color` from the theme. An example where we did it is here: https://github.com/Automattic/sensei-pro/commit/84a012cb445266e4fa99b3c6c748fe658f3f3790 (we used the same dependency that is used by Gutenberg for contrast check).

And it could be even cleaner if we could use it as the normal link style. But I suspect the designs looked better with the background, right?